### DISCUSSION TITLE
Scitos wont open

### DISCUSSION THREAD

**Discussion by sammycarey** [scientific-tool-set/scitos#37] [Q&A] [answered]
Hi there, I have a Mac computer and have downloaded the latest version of Scitos, and when I click on the app to open it, it will not work. It comes up momentarily in my dock before disappearing again.

Is there something I can do to fix this?

Thanks, Sam

**Comment by CarstenWickner** [comment, marked as answer]
Hi Sam,

TL;DR; it sounds like you're describing what I explain under "Issue 2" below. I'm just listing the other items for completeness' sake.

It's true macOS dislikes anything that is not properly validated and downloaded through the App Store.
That's why most things downloaded from the internet are deemed not trustworthy, including SciToS (since I'm unwilling to pay for the Apple Developer License one would need in order to be accepted into the App Store, let alone all the other hoops one would have to jump through. It's a free tool and it should stay that way, but I digress...).

### Issue 1: “SciToS” is damaged and can’t be opened. You should move it to the Bin.
In some cases, macOS will show this rather misleading message, which is just a harsh way of letting you know, that the application was put into "quarantine".  
However, if you do trust the application you can lift that "quarantine" via the following steps:
1. Open a Terminal (or another console). By default, this may open in your user's home folder.
2. Navigate to the directory where the unpacked `SciToS.app` is – via the `cd` command, e.g. `cd Downloads/target` if you simply unpacked it directly in your `Downloads` directory.
3. Then lift the quarantine with the following command:  
```sh
sudo xattr -r -d com.apple.quarantine SciToS.app
```
4. Enter your password to confirm the elevated access of the "super-user-do"/`sudo` command.
5. Now you can start close the Terminal and start using the SciToS.app.

### Issue 2: the SciToS.app closes immediately after start (insufficient permissions)
There may be insufficient permissions for the application to access all its parts.  
Let's fix 'em then:
1. Open a Terminal
2. Navigate to the directory where the unpacked `SciToS.app` is – via the `cd` command, e.g. `cd Downloads/target` if you simply unpacked it directly in your `Downloads` directory.
3. Then (recursively) allow all parts in the app bundle (which is really just a special directory) to be executed:  
```sh
chmod -R +x SciToS.app
```

If that still doesn't help:
1. Open a Terminal
2. Navigate to the directory where the unpacked `SciToS.app` is – via the `cd` command.
3. Then do what the double-click does:
```
open SciToS.app
```
4. Let me know what the output is after the application crash

----

If you don't want to do the above or there is still no meaningful error output, you can give up the convenience of the special macOS `.app` and just use the portable Java application directly, by downloading the `.zip` file from the [Releases](https://github.com/scientific-tool-set/scitos/releases) page and double-click the unpacked `scitos.jar` in the base folder.
If you do that though, you may encounter the next challenge.

### Issue 3: “scitos.jar” cannot be opened because it is from an unidentified developer.
Again, the downloaded file has been put into "quarantine", but the fix supposedly does not involve any Terminal commands:
1. Open your "System Preferences"
2. Select "Security & Privacy"
3. Go to the "General" tab
4. At the bottom it should say:  
> "scitos.jar" was blocked from use because it is not from an identified developer.
5. Click on the "Open Anyway" button on the right, resulting in another warning being shown:  
> macOS cannot verify the developer of "scitos.jar". Are you sure you want to open it?  
> By opening this app, you will be overriding system security which can expose your computer and personal information to malware that may harm your Mac or compromise your privacy.
6. If you still trust me, click "Open". 😉

Once that is done, you may still encounter the next challenge:

### Issue 4: the scitos.jar closes immediately after double-clicking
I know, you've been here before, but we have one more way to get this running:
1. Open a Terminal
2. Navigate to the directory where the unpacked `scitor.jar` is – via the `cd` command
3. Start the Java application directly:
```sh
java -jar scitos.jar
```
4. If that is still not working, perhaps you have no suitable Java installation after all.
Check for the currently installed version and let me know what it says:
```sh
java --version
```
Or go directly to https://adoptopenjdk.net/ and download the installer for JDK11 or newer.

----

I hope something of the above helped you.
Looking forward to your feedback.

Cheers,
Carsten

**Comment by sammycarey** [comment]
Hi Carsten,

I have just managed to get it working through the java command! Thank you very much for your quick and detailed response! It is greatly appreciated and I look forward to using it!

All the best,
Sam

**Comment by sammycarey** [comment]
Aleea also mentioned that the same command enables her to open Scitos on her windows computer.

All the best, 
Sam
